### ISSUE TITLE
bgpq4 0.0.9 fails to build on macOS

### ISSUE THREAD

**Issue by alebcay** [bgp/bgpq4#46]
`make` step of building `bgpq4` 0.0.9 (from tarball, after `bootstrap` and `configure`) fails with:
```
Making install in include
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in compat
  CCLD     libcompat.la
ar: no archive members specified
usage:  ar -d [-TLsv] archive file ...
	ar -m [-TLsv] archive file ...
	ar -m [-abiTLsv] position archive file ...
	ar -p [-TLsv] archive [file ...]
	ar -q [-cTLsv] archive file ...
	ar -r [-cuTLsv] archive file ...
	ar -r [-abciuTLsv] position archive file ...
	ar -t [-TLsv] archive [file ...]
	ar -x [-ouTLsv] archive [file ...]
make[1]: *** [libcompat.la] Error 1
make: *** [install-recursive] Error 1
```

<details>
<summary>Full build log</summary>
<pre>
==> ./bootstrap
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
aclocal: warning: couldn't open directory 'm4': No such file or directory
autoreconf: configure.ac: tracing
autoreconf: running: glibtoolize --copy --force
glibtoolize: putting auxiliary files in '.'.
glibtoolize: copying file './ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
glibtoolize: copying file 'm4/libtool.m4'
checking for dlfcn.h... yes
checking for objdir... .libs
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fno-common -DPIC
checking if clang PIC flag -fno-common -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin20.5.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking whether CC supports -Wno-pointer-sign... yes
checking if compiling with clang... yes
checking for strlcpy... yes
checking for pledge... no
checking for socket in -lsocket... no
checking for getaddrinfo in -lnsl... no
checking for sys/cdefs.h... yes
checking for sys/queue.h... yes
checking for sys/tree.h... no
checking for sys/select.h... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating include/Makefile
config.status: creating compat/Makefile
config.status: executing depfiles commands
config.status: executing libtool commands
==> make install
Making install in include
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in compat
  CCLD     libcompat.la
ar: no archive members specified
usage:  ar -d [-TLsv] archive file ...
	ar -m [-TLsv] archive file ...
	ar -m [-abiTLsv] position archive file ...
	ar -p [-TLsv] archive [file ...]
	ar -q [-cTLsv] archive file ...
	ar -r [-cuTLsv] archive file ...
	ar -r [-abciuTLsv] position archive file ...
	ar -t [-TLsv] archive [file ...]
	ar -x [-ouTLsv] archive [file ...]
make[1]: *** [libcompat.la] Error 1
make: *** [install-recursive] Error 1
</pre>
</details>

This error was encountered while packaging `bgpq4` 0.0.9 for the [Homebrew package manager](https://brew.sh). Downstream pull request at https://github.com/Homebrew/homebrew-core/pull/83515. Thank you!

**Comment by job** [comment]
Do you have a proposed solution?

**Comment by alebcay** [comment]
I don't have a patch yet (a complete novice when it comes to Automake and friends), but I did do some more digging. I think the problem is that macOS provides `strlcpy`; as a result, `libcompat` has no objects in it at all (the folder `compat/.libs` is empty), hence why `ar` complains about no archive members specified - because there's nothing to put in `libcompat.la`.

I tried building on Linux, which configure reports has no `strlcpy`, and noticed that before `CCLD libcompat.la`, there was a line `CC strlcpy.lo`. The build proceeded without any issues.

So, generally speaking, there needs to be a way to handle `libcompat.la` being empty/unneeded. Just to clarify, it looks like this issue was introduced in version 0.0.8, not in 0.0.9.
